platform/elasticsearch/client.go

```diff
@@ -24,18 +26,50 @@ type SimpleClient struct {
 	config *config.ElasticsearchConfiguration
 }
 
-func NewSimpleClient(configuration *config.ElasticsearchConfiguration) *SimpleClient {
-	client := &http.Client{
+func NewHttpsClient(configuration *config.ElasticsearchConfiguration, timeout time.Duration) *http.Client {
+	tlsConfig := &tls.Config{
+		MinVersion:         tls.VersionTLS12,
+		InsecureSkipVerify: true,
+	}
+
+	if configuration.CACertPath != "" {
+		caCert, err := os.ReadFile(configuration.CACertPath)
+		if err != nil {
+			panic(fmt.Sprintf("failed to read CA certificate: %v", err))
```

Comment:
We don't want `panic`s in production code. 
I think logging at `Error` level and fallback to certificate-less client would be more appropriate.

After all, the server will still throw handshake error, yet at least the service won't crash or bootloop in Cloud/k8s-based environments.

Reply:
done.